### PR TITLE
Chore - exam participation cover tests

### PR DESCRIPTION
**Pull request by FuchsDominik**
### Checklist

- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)

### Motivation and Context

- exam client test improvements

### Description

- Implemented client tests
- Removed a TODO which was from July and is probably not going to get updated?
- Removed an else branch which is not reachable, since in a test run, the run automatically starts. Code-wise this means the hasStarted() always returns true and therefore, the else clause can not be reached. For more details see the code changes (line 130 and 132 are decisive).

### Steps for Testing

1. Log in to Artemis
2. Create an exam and participate in it (using a student run and using a test run)
3. Make sure it works properly (the cover page mainly)

### Test Coverage

exam-participation-cover.component.ts: 100%

### CODE REVIEW

**Review by airelawaleria — APPROVED** [review]
Tested in the testing session on ts1. Works as expected

**Review by JohannesWeiss — APPROVED** [review]
Tested on ts1 in testing session, works as expected. Approve.

**Review by simonzilker — APPROVED** [review]
everything worked as usual on ts1

**Review by marvinchimeka — APPROVED** [review]
tested in testing session. Approve

File: src/main/webapp/app/exam/participate/exam-cover/exam-participation-cover.component.ts [hunk 1]
```diff
@@ -129,14 +129,7 @@ export class ExamParticipationCoverComponent implements OnInit, OnDestroy {
...
             this.examParticipationService.saveStudentExamToLocalStorage(this.exam.course!.id!, this.exam.id!, this.studentExam);
-            if (this.hasStarted()) {
-                this.onExamStarted.emit(this.studentExam);
-            } else {
-                this.waitingForExamStart = true;
-                this.interval = window.setInterval(() => {
-                    this.updateDisplayedTimes(this.studentExam);
-                }, 100);
-            }
+            this.onExamStarted.emit(this.studentExam);
```
Comment (krusche) [inline src/main/webapp/app/exam/participate/exam-cover/exam-participation-cover.component.ts:132]: @FuchsDominik why did you change the implementation here? are you sure the logic is not necessary for a test run?

Comment (FuchsDominik) [reply src/main/webapp/app/exam/participate/exam-cover/exam-participation-cover.component.ts:132]: During testing i discovered that to get into this branch, the exam run has to be a test run. (See line 130) hasStarted() automatically returns true if the exam is in a test run, that's why it is not possible that the check in line 132 return false. Therefore the else clause will never get executed. @anditurdiu confirmed my findings, right @anditurdiu ?

Comment (FuchsDominik) [reply src/main/webapp/app/exam/participate/exam-cover/exam-participation-cover.component.ts:132]: Logic-wise, an instructor does not have to wait for the exam test run as the test run starts automatically

Comment (krusche) [reply src/main/webapp/app/exam/participate/exam-cover/exam-participation-cover.component.ts:132]: ok sounds good

**Review by krusche — APPROVED** [review]
